src/adapter/odata/src/lib.rs

```diff
@@ -0,0 +1,390 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use axum::async_trait;
+use datafusion::arrow::datatypes::{Schema, SchemaRef};
+use datafusion::dataframe::DataFrame;
+use datafusion_odata::collection::{QueryParams, QueryParamsRaw};
+use datafusion_odata::context::{CollectionContext, ServiceContext};
+use dill::Catalog;
+use kamu_core::*;
+use opendatafabric::*;
+
+///////////////////////////////////////////////////////////////////////////////
+
+// TODO: Externalize config
+const DEFAULT_RECORDS_PER_PAGE: usize = 100;
+const MAX_RECORDS_PER_PAGE: usize = usize::MAX;
+
+///////////////////////////////////////////////////////////////////////////////
+// Router
+///////////////////////////////////////////////////////////////////////////////
+
+pub fn router_single_tenant() -> axum::Router {
+    axum::Router::new()
+        .route("/", axum::routing::get(odata_service_handler_st))
+        .route("/$metadata", axum::routing::get(odata_metadata_handler_st))
```

Comment:
Dollar symbol in URL?

Reply:
yep, that's OData spec for you...
they also use dollar sign for query parameters:
```
/odf/account/dataset?$orderby=offset?$top=10
```